Highcharts conversion to Plotly JS (#1822)
ubccr/xdmod: CHANGELOG.md

# Open XDMoD Change Log

## 2024-XX-XX v11.0.0
- Important Notes

>     - Open XDMoD is now no longer bundled with any non-commercial licenses. The charting library used in Open XDMoD has changed from [Highcharts](https://www.highcharts.com/) to [Plotly JS](https://plotly.com/javascript/), an open source library. This transition removes the non-commercial license required from the Highcharts library. Please refer to the [license notices](https://open.xdmod.org/10.5/notices.html) for more information about the open source licenses bundled with Open XDMoD.

Reviewer: ```suggestion
    - Open XDMoD is now no longer bundled with any non-commercial licenses. The charting library used in Open XDMoD has changed from [Highcharts](https://www.highcharts.com/) to [Plotly JS](https://plotly.com/javascript/), an open source library. This transition removes the non-commercial license required from the Highcharts library. Please refer to the [license notices](https://open.xdmod.org/11.0/notices.html) for more information about the open source licenses bundled with Open XDMoD.
```

- New Features
    - General
        - Chart axes are draggable to adjust the chart range. For charts in the Metric Explorer, the axis range can be reset to the default through the axis context menu. For other charts, the chart can be reset to the default by double-clicking.

>         - Rendering support for a subset of HTML tags in a chart's main title. Please refer to the [list](https://plotly.com/javascript/reference/layout/annotations/#layout-annotations-items-annotation-text) of supported HTML tags. Unsupported tags will not render but still display in the title.

Reviewer: I would actually list the tags here since it is a small list and because the documentation at that URL can and will change without our 11.0 source code syncing to it.

>         - Rendering support for a subset of HTML tags in a chart's main title. Please refer to the [list](https://plotly.com/javascript/reference/layout/annotations/#layout-annotations-items-annotation-text) of supported HTML tags. Unsupported tags will not render but still display in the title.

Reviewer: ```suggestion
        - Rendering support for a subset of HTML tags in a chart's main title. Please refer to the [list](https://plotly.com/javascript/reference/layout/annotations/#layout-annotations-items-annotation-text) of supported HTML tags. Unsupported tags will render as plain text instead of HTML.
```

>         - Rendering support for a subset of HTML tags and entities in a chart's main title. Unsupported tags/entities will render as plain text instead of HTML. The following HTML tags are supported: &lt;br&gt;, &lt;b&gt;, &lt;i&gt;, &lt;sup&gt;, and &lt;sub&gt;. The following entities are supported: Greek letters (recommended Unicode format, e.g., &amp;#x3BC; for lowercase mu), some mathematical symbols (e.g., &plusmn; for ±), and degrees (&deg;).

Reviewer: ```suggestion
        - Rendering support for a subset of HTML tags and entities in a chart's main title. Unsupported tags/entities will render as plain text instead of HTML. The following HTML tags are supported: `<br>`, `<b>`, `<i>`, `<sup>`, and `<sub>`. The following entities are supported: Greek letters (recommended Unicode format, e.g., `&#x3BC;` for lowercase mu), some mathematical symbols (e.g., `&plusmn;` for ±), and degrees (`&deg;`).
```

- Uncategorized
    - Chart Feature Changes
        - For line charts, the context menu for a data series is brought up by clicking on its points, not its lines. As before, the context menu can also be brought up by clicking on it in the legend.
        - The button for resetting the chart zoom in Metric Explorer charts appears in the chart context menus instead of a button on the chart. Resetting the chart zoom for charts in all other tabs now requires a double click on the plotting area instead of a button on the chart.

>     - Chart Features Temporarily Removed

Reviewer: ```suggestion
    - Chart Features Removed
```
I'm going back on my previous suggestion since I think we shouldn't suggest these will be put back in unless we have a roadmap for doing so.

# Open XDMoD Change Log

## 2024-XX-XX v11.0.0
- Important Notes
    - Open XDMoD is now no longer bundled with any non-commercial licenses. The charting library used in Open XDMoD has changed from [Highcharts](https://www.highcharts.com/) to [Plotly JS](https://plotly.com/javascript/), an open source library. This transition removes the non-commercial license required from the Highcharts library. Please refer to the [license notices](https://open.xdmod.org/11.0/notices.html) for more information about the open source licenses bundled with Open XDMoD.
- New Features
    - General
        - Chart axes are draggable to adjust the chart range. For charts in the Metric Explorer, the axis range can be reset to the default through the axis context menu. For other charts, the chart can be reset to the default by double-clicking.

>         - Rendering support for a subset of HTML tags in a chart's main title. Unsupported tags will render as plain text instead of HTML. The following HTML tags are supported:
>             - Newline
>             - Bold text
>             - Italic text
>             - Superscripts
>             - Subscripts
>             - Hyperlinks
>             - Greek Letters (Recommend unicode format i.e. '$#x3BC;' for lowercase mu)
>             - Some math symbols (such as ±)
>             - Degrees

Reviewer: ```suggestion
        - Rendering support for a subset of HTML tags and entities in a chart's main title. Unsupported tags/entities will render as plain text instead of HTML. The following HTML tags are supported: <br>, <b>, <i>, <sup>, <sub>, and <a>. The following entities are supported: Greek letters (recommended Unicode format, e.g., &#x3BC; for lowercase mu), some mathematical symbols (e.g., &plusmn; for ±), and degrees (&deg;).
```
